src/main/java/com/atlassian/bitbucket/jenkins/internal/credentials/CredentialUtils.java

```diff
@@ -29,6 +30,9 @@ public static Credentials getCredentials(@Nullable String credentialsId) {
         if (creds == null) {
             creds = getCredentials(UsernamePasswordCredentials.class, credentialsId);
         }
+        if (creds == null) {
```

Comment:
These null checks are quite messy. I just tacked one on at the end, but we could probably wrap this in an optional (and even merge the two methods given the second is private?)

Reply:
Did it!